helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -2061,4 +2069,43 @@ public ZKHelixAdmin build() {
               _zkAddress), false);
     }
   }
+
+  private Map<String, InstanceConfig> findTimeoutOfflineInstances(String clusterName,
+      Long timeout) {
+    Map<String, InstanceConfig> instanceConfigMap = new HashMap<>();
+    // in case there is no customized timeout value, use the one defined in cluster config
+    if (timeout == null) {
+      timeout = _configAccessor.getClusterConfig(clusterName).getOfflineNodeTimeOutForPurge();
+    }
```

Comment:
instead of doing the `null` check, maybe you could define a `NOT_SET` constant of some sort...